lib/module.js

```diff
@@ -9,10 +9,13 @@ const dnt = "(function(w,n,d,m,e,p){w[d]=(w[d]==1||n[d]=='yes'||n[d]==1||n[m]==1
 module.exports = async function gtmModule (_options) {
   requireNuxtVersion(this.nuxt, '2.12.0')
 
+  const runtimeConfig = (this.$config && this.$config.gtm) || {}
```

Comment:
We don't have `this.$config` in a module context (sorry if my previous comments were misleading)

For module runtime support we need both changes like this:

- https://github.com/nuxt/http/blob/dev/lib/module.js#L90
- https://github.com/nuxt/http/blob/dev/lib/plugin.js#L186

If this makes PR complicated, i would be also okay with reverting to initial implementation for secondary `id` support via runtimeConfig :)

Reply:
🤔 I tested this implementation locally and worked just fine. I'll check your links later today.

_(your comments are awesome, I'm learning a lot with you. It's been a pleasure to contribute with these projects)_

Reply:
`this` inside module export is `ModuleContainer` ([src](https://github.com/nuxt/nuxt.js/blob/dev/packages/core/src/module.js) it is almost impossible working :D